algorithms/unit_tests/TestStdAlgorithmsTeamFindIf.cpp

```diff
@@ -0,0 +1,238 @@
+        m_distancesView(distancesView),
+        m_apiPick(apiPick) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+    auto myRowViewFrom = Kokkos::subview(m_dataView, myRowIndex, Kokkos::ALL());
+    const auto val     = m_greaterThanValuesView(myRowIndex);
+    GreaterEqualFunctor unaryOp{val};
+
+    switch (m_apiPick) {
+      case 0: {
+        auto it = KE::find_if(member, KE::cbegin(myRowViewFrom),
+                              KE::cend(myRowViewFrom), unaryOp);
+
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+
+      case 1: {
+        auto it = KE::find_if(member, myRowViewFrom, unaryOp);
+
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(const bool predicatesReturnTrue, std::size_t numTeams,
+            std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level find_if
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  const ValueType lowerBound{5}, upperBound{523};
+  auto [dataView, dataViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols, Kokkos::pair{lowerBound, upperBound},
+      "dataView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // find_if returns an iterator so to verify that it is correct
+  // each team stores the distance of the returned iterator from the
+  // beginning of the interval that team operates on and then we check
+  // that these distances match the std result
+  Kokkos::View<std::size_t*> distancesView("distancesView", numTeams);
+
+  // If predicatesReturnTrue == true, we want to ensure that eventually, for
+  // some value from each of dataView's row, predicate GreaterEqualFunctor will
+  // return true. To do that, for each numTeams, a random j index from a range
+  // [0, numCols) is used to obtain a value from dataView, and later that value
+  // is used for creating concrete GreaterEqualFunctor predicate.
+  //
+  // If searchedValuesExist == false we want to ensure the opposite, so every
```

Comment:
comments needs to be revised, searchedValuesExist is not defined anywhere

Reply:
Revised. And other tests checked as well.